src/gatsby-node.js

```diff
@@ -302,6 +306,43 @@ exports.sourceNodes = async (
         );
       }
     };
+    ws.onclose = function() {
+      reconnectingWebSocket(ws, previousDelay, timeoutId);
+    };
+    ws.onerror = function(err) {
+      reporter.error(err.stack);
+    };
+
+    ws.on('pong', () => {
+      count--;
+    });
+
+    const intervalId = setInterval(() => {
+      if (ws.readyState == 1) {
+        ws.ping('heartBeat');
+        count++;
+      }
+      if (count == 2) {
+        count = 0;
+        reconnectingWebSocket(ws, previousDelay, timeoutId);
+      }
+    }, 60000);
+  };
+
+  const reconnectingWebSocket = function(ws, previousDelay, timeoutId) {
+    previousDelay = Math.min(60, previousDelay * (2 - Math.random()));
+    if (timeoutId) {
```

Comment:
Isn’t this function already called from this timeout so that the timeout has already fired and there is no longer need to clear it? Or is there real use for the timeoutid argument?

Reply:
Just one more time overthinking. I am thinking that might be there are possibilities that webSocketStart takes more time to execute during this time we can have multiple timeouts. so, I cleared that. But I think that there is no need for timeoutId argument. I will remove this after your full review.

Reply:
I removed the timeoutId in latest commit :)